backend/backend-schema-proposal.md

```diff
@@ -39,59 +39,128 @@ GET `/request/{van_id}`
 GET `/location`
 
 - Subscribes the client to SSE triggered by new coordinates to any van
-- Responses are in the format `(lat, lon, route_id, van_id, timestamp)` where
-  `van_id` & `route_id` are 8-bit ints, `lat` & `lon` are 64-bit doubles, and
-  `timestamp` is a UNIX timestamp
 - Any response with an earlier timestamp than the most recent should be discarded
 - Any timestamp more than 1 min in the future should be discarded
 - Communication can be HTTP or HTTPS, but HTTPS is already being used
+- Response body is defined in JSON as per the following typescript spec:
+
+``ts
+interface Location {
+    vanId: number,
+    routeId: number,
+    location: Coordinate,
+    timestampMillis: number
+}
+``
 
 GET `/routes`
 
 - Retrieves a predefined list of routes and their waypoints
 - Coordinates are defined in a config file
-- Responses are a JSON array of routes in the following format:
+- Response body is defined in JSON as per the following typescript spec:
+
+```typescript
+interface Routes {
+    routes: Route[],
+    stops: Stop[]
```

Comment:
why are routes not connected to their stops?

Reply:
Two reasons:
1. Multiple routes can share one stop. So, we can avoid fetching redundant data if we separate them.
2. By separating them, the frontend can draw the path of the routes, and then overlay stop icons on stop of them. If they were combined, we would have to sift through the waypoint data to identify stops, which is business logic that we don't want on the frontend.